tests/exhaustive_tests.rs

```diff
@@ -0,0 +1,46 @@
+extern crate unicode_xid;
+use unicode_xid::UnicodeXID;
+/// A `char` in Rust is a Unicode Scalar Value
+///
+/// See: http://www.unicode.org/glossary/#unicode_scalar_value
+fn all_valid_chars() -> impl Iterator<Item = char> {
+    (0u32..=0xD7FF).chain(0xE000u32..=0x10FFFF).map(|u|
+        core::convert::TryFrom::try_from(u)
+            .expect("The selected range should be infallible if the docs match impl")
+    )
+
+}
+
+#[test]
+fn all_valid_chars_do_not_panic_for_is_xid_start() {
+    for c in all_valid_chars() {
+        let _ = UnicodeXID::is_xid_start(c);
+    }
+}
+
+#[test]
+fn all_valid_chars_do_not_panic_for_is_xid_continue() {
+    for c in all_valid_chars() {
+        let _ = UnicodeXID::is_xid_continue(c);
+    }
+}
+
+fn all_invalid_chars() -> impl Iterator<Item = char> {
+    (0xD800u32..0xE000).chain(0x110000u32..=core::u32::MAX)
+        .map(|u| unsafe { core::char::from_u32_unchecked(u)})
```

Comment:
That's immediate UB: Rust `char` can't hold something which is not unicode scalar value.

In other words, if `UnicodeXID::is_xid_start` ever sees invalid char:

* that means that some *other* code has UB
* it's valid for `rustc` to make `is_xid_start` behave arbitrary on this code. In particular, panicking would be theoretically admissible behavior. 

I suggest we just drop this test, or write a comment explaining *why* we do this.

Reply:
I agree, the value of the `invalid` inputs tests is pretty small. Like @matklad said, panicking is theoretically admissible since we're dealing in UB.  Said `invalid` input tests only demonstrate that while the code might theoretically panic given a `char` generated through a particular form of UB, for the given set of compilation circumstances exercised here, it appears not to panic.

That's narrow enough bit of assurance that I'm fine taking it out.  Follow-up commit incoming.